shopfloor_reception/services/reception.py

```diff
@@ -356,26 +365,44 @@ def _set_quantity__by_package(self, picking, selected_line, barcode):
         search = self._actions_for("search")
         package = search.package_from_scan(barcode)
         if package:
-            dest_location = selected_line.location_dest_id
-            child_locations = self.env["stock.location"].search(
-                [("id", "child_of", dest_location.id)]
-            )
             pack_location = package.location_id
             if pack_location:
-                if pack_location not in child_locations:
+                (
+                    move_dest_location_ok,
+                    pick_type_dest_location_ok,
+                ) = self._check_location_ok(pack_location, selected_line, picking)
+                if not (move_dest_location_ok or pick_type_dest_location_ok):
                     # If the scanned package has a location that isn't a child
                     # of the move dest, return an error
                     message = self.msg_store.dest_location_not_allowed()
                     return self._response_for_set_quantity(
                         picking, selected_line, message=message
                     )
-                else:
-                    # If the scanned package has a valid destination,
-                    # set both package and destination on the package,
-                    # and go back to the selection line screen
-                    selected_line.result_package_id = package
-                    selected_line.location_dest_id = pack_location
-                    return self._response_for_select_move(picking)
+                quantity = selected_line.qty_done
+                __, qty_check = selected_line._split_qty_to_be_done(
```

Comment:
I'm kind of worried about that. If qty_check is "lesser", then qty_done has been altered on self, it created a new line, and we do nothing about this.

Reply:
If `qty_check` is "lesser", we proceed with the workflow and we post a new line, which is the expected behavior right?

Reply:
@jbaudoux what do you think about this?

Reply:
Looks all good.
If qty done is 2 and todo is 3, then you split the line in 2. On the line with qty done = qty todo = 2, you then write the package & location. The other new line (qty done = 0 and qty todo = 1) will be available when you start over the process.

Reply:
All good then :blush: